src/esthri/src/lib.rs

```diff
@@ -759,27 +741,39 @@ async fn head_object_request<T>(s3: &T, bucket: &str, key: &str) -> Result<Optio
 where
     T: S3 + Send,
 {
-    let res = handle_dispatch_error(|| async {
-        let hor = HeadObjectRequest {
-            bucket: bucket.into(),
-            key: key.into(),
-            ..Default::default()
-        };
+    let mut res = Some(
+        handle_dispatch_error(|| async {
+            let hor = HeadObjectRequest {
+                bucket: bucket.into(),
+                key: key.into(),
+                ..Default::default()
+            };
 
-        s3.head_object(hor).await
-    })
-    .await;
+            s3.head_object(hor).await
+        })
+        .await,
+    );
 
-    match res {
-        Ok(hoo) => process_head_obj_resp(hoo),
-        Err(RusotoError::Unknown(e)) => {
+    match res.as_mut() {
+        Some(Ok(_)) => {
+            let hoo = res.unwrap().unwrap();
+            process_head_obj_resp(hoo)
+        }
+        Some(Err(RusotoError::Unknown(e))) => {
```

Comment:
This is a bit nuts, in order to deal with the borrow checker, I need to wrap this in an `Option` so that I would borrow `e` (a sub-component of the `res` object) then move that object later when constructing the wrapped error.  I thought I would need to use `Option::take` but it turned out to be unnecessary, not sure why.